docs/using.rst

```diff
@@ -39,3 +39,20 @@ To use this, you should modify your ``setup.py`` file to use
 Note that if you use this, extension-helpers will also we create a
 ``packagename.compiler_version`` submodule that contain information about the
 compilers used.
+
+Python limited API
+------------------
+
```

Comment:
In a similar vein to my other comment, it would be nice to explain when using this is appropriate and when it isn't as I expect most people don't even know this exists (I didn't) and why they would want to use it.

Reply:
Yes, you're right! It's not a well-known feature, although it seems to be well supported by pip. Part of the aim of this PR is to raise awareness by making it easy for Astropy affiliated packages to opt in.